stdlib/runtime/numbers.gr

```diff
@@ -520,9 +520,31 @@ let numberEqualFloat64Help = (x, y) => {
   }
 }
 
+/*
 let numberEqualFloat32Help = (x, y) => {
   numberEqualFloat64Help(WasmF64.promoteF32(x), y)
 }
+*/
+
+let numberEqualFloat32Help = (x, y) => {
```

Comment:
It's required for the new NaN constant to work correctly. I'd go as far as to say it's a bugfix, so I am not sure how worried we should be about it being breaking (since the current behavior is incorrect).